src/Core/PlaybackFormat.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of littleredbutton/bigbluebutton-api-php.
+ *
+ * littleredbutton/bigbluebutton-api-php is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * littleredbutton/bigbluebutton-api-php is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with littleredbutton/bigbluebutton-api-php. If not, see <http://www.gnu.org/licenses/>.
+ */
+namespace BigBlueButton\Core;
+
+class PlaybackFormat
+{
+
+    /** @var string */
+    private $type;
+
+    /** @var string */
+    private $url;
+
+    /** @var int */
+    private $processingTime;
+
+    /** @var int */
+    private $length;
+
+    /** @var array */
+    private $imagePreviews;
+
+    /** @var \SimpleXMLElement */
+    private $imagePreviewsRaw;
+
+    public function __construct(\SimpleXMLElement $xml)
+    {
+        $this->type               = $xml->type->__toString();
+        $this->url                = $xml->url->__toString();
+        $this->processingTime     = (int) $xml->processingTime->__toString();
+        $this->length             = (int) $xml->length->__toString();
+
+        $this->imagePreviewsRaw   = $xml->preview->images;
+    }
+
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    public function getUrl(): string
+    {
+        return $this->url;
+    }
+
+    public function getProcessingTime(): int
+    {
+        return $this->processingTime;
+    }
+
+    public function getLength(): int
+    {
+        return $this->length;
+    }
+
+    public function hasImagePreviews(): bool
+    {
+        return !!$this->imagePreviewsRaw;
+    }
+
+    /**
+     *
+     * @return array<int, array{width: int, height: int, alt: string, url: string}>
+     */
+    public function getImagePreviews(): array
```

Comment:
Instead of a set of associative arrays, we could create a set of `ImagePreview` objects here. This would improve the usability, typing and interface very much:

~~~php
class ImagePreview
{
    private $width;
    private [...];
}
~~~

With respective getters, of course. The class would take all properties only via constructor and would immutable afterwards, i.e. having only the (typed) getters.

Reply:
That was what I was taking about :smile:. Just thought it's maybe a bit overkill, but yeah you are right, the readability will benefit from wrapping it into classes.

Reply:
@sualko Any progress?

Reply:
> That was what I was taking about smile. Just thought it's maybe a bit overkill, but yeah you are right, the readability will benefit from wrapping it into classes.

There is a simple rule: You can never have enough data objects in your code ;) .

Reply:
@sualko If you haven't the time now, I would be happy and to finish here by resolving this discussion. Would that be fine for you?

Reply:
@FelixJacobi thanks for the offer. I just implemented it by myself :smile:. This means we can probably merge it.